src/pages/Storefront/Storefront.js

```diff
@@ -16,10 +17,38 @@ const Store = () => {
     createCheckout();
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
+  const { error: getAllTeamsError, data: getAllTeamsData } = useQuery(GET_ALL_TEAMS);
```

Comment:
While this works, does it make sense to get ALL the teams every time instead of only the team we care for? Even more specifically, why should we bother fetching any teams at all if there's no query parameter present in the first place?

I would suggest something more along these lines:

```
  const [getTeam, { loading: getTeamLoading, error: getTeamError, data: getTeamData }] = useLazyQuery(GET_TEAM_INFO);
  const { search } = useLocation();

  useEffect(() => {
    if (search) {
      const queries = new URLSearchParams(search);
      const teamId = queries.get('team');
      if (teamId) {
        getTeam({ variables: { id: teamId } });
      }
    }
  }, [search, getTeam]);
```

`useLocation` is from `react-router`, and you can see https://www.apollographql.com/docs/react/data/queries/#executing-queries-manually for more details on `useLazyQuery`! 

You could then probably pass `getTeamLoading`, `getTeamData` and `getTeamError` to `TeamBanner` and have the same functional logic

Disclaimer: idk if my code actually works, just a quick sample I threw together

Reply:
Fixed!